ludwig/trainers/trainer.py

```diff
@@ -1200,6 +1202,9 @@ def _train_loop(
                     if self.is_coordinator():
                         progress_tracker.save(os.path.join(save_path, TRAINING_PROGRESS_TRACKER_FILE_NAME))
```

Comment:
There's another place where the checkpoint manager saves the model, outside/after 

```
should_break = self._train_loop
```

Should we add this callback there as well?

Also, I'm noticing that this code may be duplicated in both places -- would you be able to refactor into a shared library?